zebra-state/src/on_disk/block_index.rs

```diff
@@ -0,0 +1,163 @@
+use std::{
+    collections::{btree_map::Entry, BTreeMap, HashMap},
+    sync::{Arc, Weak},
+};
+use zebra_chain::serialization::{ZcashDeserialize, ZcashSerialize};
+use zebra_chain::{
+    block::{Block, BlockHeaderHash},
+    types::BlockHeight,
+};
+
+type Error = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+pub(super) struct BlockIndex {
+    cache: WeakCache,
+    storage: sled::Db,
+}
+
+impl Default for BlockIndex {
+    fn default() -> Self {
+        let config = sled::Config::default();
```

Comment:
how much configurability do we want to expose to users? My inclination is to get the os's default config directory and create a `zebra-state` folder there, though this needs to be at least somewhat configurable to let unit tests control where the DB gets instantiated.

for reference, these are the options we could expose: https://docs.rs/sled/0.31.0/sled/struct.Config.html